multiview/embed/splitae.py

```diff
@@ -0,0 +1,206 @@
+            if validationXs is not None:
+                testError = self._testError(validationXs)
+                print("Average test  error during epoch {} was {}\n"
+                      .format(epoch, testError)) if printInfo else None
+                epochTestErrors.append(testError)
+
+        if printInfo:
+            plt.plot(epochTrainErrors, label="train error")
+            if validationXs is not None:
+                plt.plot(epochTestErrors, label="test error")
+            plt.title("Errors during training")
+            plt.xlabel("Epoch")
+            plt.ylabel("Error")
+            plt.legend()
+            plt.show()
+
+    def _testError(self, Xs):
+        device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        nSamples = Xs[0].shape[0]
+        validationBatchSize = self.batchSize
+        testIndices = np.random.choice(nSamples, validationBatchSize,
+                                       replace=False)
+        view1Batch = torch.FloatTensor(Xs[0][testIndices])
+        view2Batch = torch.FloatTensor(Xs[1][testIndices])
+        with torch.no_grad():
+            embedding = self.view1Encoder(view1Batch.to(device))
+            view1Reconstruction = self.view1Decoder(embedding)
+            view2Reconstruction = self.view2Decoder(embedding)
+            view1Error = torch.nn.MSELoss()(view1Reconstruction,
+                                            view1Batch.to(device))
+            view2Error = torch.nn.MSELoss()(view2Reconstruction,
+                                            view2Batch.to(device))
+            totalError = view1Error + view2Error
+        return totalError.item()
+
+    def transform(self, Xs):
+        """
+        Transform the given view with the trained autoencoder.
+        Parameters
+        ----------
+        Xs: a list with one array representing the View1 view of some data.
+        The array must have the same number of columns (features) as
+        the View1 presented in the `fit(...)` step.
+        Returns
+        ----------
+        embedding: the embedding of the View1 data
+        view1Reconstruction: the reconstructed View1
+        view2Prediction: the predicted View2
+        """
+        device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        view1 = torch.FloatTensor(Xs[0])
+        with torch.no_grad():
+            embedding = self.view1Encoder(view1.to(device))
+            view1Reconstruction = self.view1Decoder(embedding)
+            view2Prediction = self.view2Decoder(embedding)
+        return (embedding.cpu().numpy(), view1Reconstruction.cpu().numpy(),
+                view2Prediction.cpu().numpy())
+
+    def fit_transform(self, Xs):
+        """
+        `fit(Xs)` and then `transform(Xs[:1])`. Note that this method will be
+        embedding data that the autoencoder was trained on.
+        Parameters:
+        ----------
+        Xs: see `fit(...)` Xs parameters
```

Comment:
same as above

Reply:
Have changed the other docstring wordings to be more consistent! Would rather keep this one as is to prevent duplicating text